pkg/docker/docker.go

```diff
@@ -439,7 +538,7 @@ func (d *stiDocker) CheckAndPullImage(name string) (*docker.Image, error) {
 	}
 
 	image, err := d.CheckImage(name)
-	if err != nil && err.(errors.Error).Details != docker.ErrNoSuchImage {
+	if err != nil && dockerapi.IsErrImageNotFound(err) {
```

Comment:
Yeah I mentioned this briefly in scrum and I thought in some of the comments near the top of docker_test.go - the engine-api error types themselves are not public, only these methods which allow you to discern the type of error.  So unlike before with unit tests with go-dockerclient, we can't instantiate the specific error that will pass or fail their helper method in the fashion we would like.  Is there another alternative I'm missing.

Aside from the test question - code will be updated in next push.